src/shared/sleep-config.c

```diff
@@ -268,39 +280,48 @@ static int read_resume_files(char **ret_resume, uint64_t *ret_resume_offset) {
                         return log_error_errno(r, "Failed to parse value in /sys/power/resume_offset \"%s\": %m", resume_offset_str);
         }
 
-        if (resume_offset > 0 && streq(resume, "0:0")) {
+        if (resume_offset > 0 && resume == 0) {
                 log_debug("Found offset in /sys/power/resume_offset: %" PRIu64 "; no device id found in /sys/power/resume; ignoring resume_offset",
                           resume_offset);
                 resume_offset = 0;
         }
 
-        *ret_resume = TAKE_PTR(resume);
+        *ret_resume = resume;
         *ret_resume_offset = resume_offset;
 
         return 0;
 }
 
-static bool location_is_resume_device(const HibernateLocation *location, const char *sys_resume, const uint64_t sys_offset) {
-        assert(location);
-        assert(location->resume);
-        assert(sys_resume);
+/*
+ * Determine if the HibernateLocation matches the resume= (device) and resume_offset= (file).
+ */
+static bool location_is_resume_device(const HibernateLocation *location, const dev_t sys_resume, const uint64_t sys_offset) {
```

Comment:
please drop the const from non-pointer parameters, that has no effect... (i.e. leave the one for location, drop the one from sys_resume and sys_offset)